src/hotspot/cpu/ppc/macroAssembler_ppc.cpp

```diff
@@ -4045,42 +4046,55 @@ void MacroAssembler::lightweight_unlock(Register obj, Register hdr, Label& slow)
     stop("Header is not fast-locked");
     bind(hdr_ok);
   }
-  Register t1 = hdr; // Reuse in debug build.
+#endif
+
+  Register temp = (t1 != noreg) ? t1 : hdr; // Reuse if no temp reg given.
+  lwz(temp, in_bytes(JavaThread::lock_stack_top_offset()), R16_thread);
+
+#ifdef ASSERT
   {
     // The following checks rely on the fact that LockStack is only ever modified by
     // its owning thread, even if the lock got inflated concurrently; removal of LockStack
     // entries after inflation will happen delayed in that case.
 
     // Check for lock-stack underflow.
     Label stack_ok;
-    lwz(t1, in_bytes(JavaThread::lock_stack_top_offset()), R16_thread);
-    cmplwi(CCR0, t1, LockStack::start_offset());
+    cmplwi(CCR0, temp, LockStack::start_offset());
     bgt(CCR0, stack_ok);
     stop("Lock-stack underflow");
     bind(stack_ok);
   }
-  {
-    // Check if the top of the lock-stack matches the unlocked object.
+#endif
+
+  // Check if the top of the lock-stack matches the unlocked object.
+  addi(temp, temp, -oopSize);
+  if (may_be_unordered) {
```

Comment:
I guess there is no need to call the slow-path here. Simply don't emit the assertion should be good enough, right?